airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/output/ExceptionHandler.kt

```diff
@@ -45,4 +46,23 @@ class ExceptionHandler(val classifiers: List<ExceptionClassifier>) {
                     .withMessage(classified.displayMessage ?: e.message)
         }
     }
+
+    fun handleCheckFailure(e: Throwable): Pair<AirbyteErrorTraceMessage, AirbyteConnectionStatus> {
+        val errorTraceMessage: AirbyteErrorTraceMessage = handle(e)
+        errorTraceMessage.failureType = AirbyteErrorTraceMessage.FailureType.CONFIG_ERROR
+        val connectionStatusMessage: String =
+            String.format(COMMON_EXCEPTION_MESSAGE_TEMPLATE, errorTraceMessage.message)
+
+        return Pair(
+            errorTraceMessage,
+            AirbyteConnectionStatus()
+                .withMessage(connectionStatusMessage)
+                .withStatus(AirbyteConnectionStatus.Status.FAILED)
+        )
+    }
+
+    companion object {
+        const val COMMON_EXCEPTION_MESSAGE_TEMPLATE: String =
+            "Could not connect with provided configuration. Error: %s"
+    }
```

Comment:
This change to the core module is fine in and of itself but this PR is incomplete as is; please either:
- also refactor the CheckOperation in the extract module to have it use `handleCheckFailure`;
- or inline this code in the CheckOperation in the load module

Otherwise we're going to forget to do this. This should be straightforward either way.

On another note, and out of scope of this PR, it's very possible that this connection status message should be revisited: in the past all CHECK ever did was try to connect to the source or the destination but AFAIK we intend to also try to create temp tables in the destination and other more advanced checks. In fact for source-firetruck we already do such advanced checks. This message isn't completely wrong but it's misleading, as the connector might be able to connect just fine and rather might instead not be happy with what it's seeing.

Reply:
Yes, I wasn't sure if you wanted me to migrate it. Will do!

Reply:
Done